tests/unit/test_types.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include <random>
+#include "gtest/gtest.h"
+#include "VecSim/types/float16.h"
+
+#ifdef OPT_AVX512_FP16
+class FP16Type : public ::testing::Test {};
+
+TEST_F(FP16Type, Test_Float16VSvecsim) {
+    static_assert(sizeof(_Float16) == sizeof(vecsim_types::float16));
+    size_t runs = 10;
+
+    std::mt19937 gen{};
+    std::uniform_real_distribution<float> dis(-0.999999, 0.999999);
+
+    for (size_t i = 0; i < runs; i++) {
+        float val = dis(gen);
+        _Float16 nativeCast = static_cast<_Float16>(val);
+        vecsim_types::float16 vecsimCast = vecsim_types::FP32_to_FP16(val);
+
+        uint16_t nativeBits, vecsimBits;
+        std::memcpy(&nativeBits, &nativeCast, sizeof(_Float16));
+        std::memcpy(&vecsimBits, &vecsimCast, sizeof(vecsim_types::float16));
+
+        ASSERT_EQ(nativeBits, vecsimBits) << "failed at iteration " << i << "with value " << val;
```

Comment:
can't it flake?

Reply:
why? we expect to get identical results

Reply:
and if we don't?

Reply:
Then we are lucky that we have this test